beacon-chain/core/helpers/payload_attestation.go

```diff
@@ -74,20 +74,18 @@ func GetPayloadTimelinessCommittee(ctx context.Context, state state.ReadOnlyBeac
 	if state.Version() < version.EPBS {
 		return nil, errPreEPBSState
 	}
-	epoch := slots.ToEpoch(slot)
-	activeCount, err := ActiveValidatorCount(ctx, state, epoch)
+	committees, err := BeaconCommittees(ctx, state, slot)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not compute active validator count")
+		return nil, errors.Wrap(err, "could not get beacon committees")
 	}
-	committeesPerSlot, membersPerCommittee := PtcAllocation(activeCount)
-	for i := uint64(0); i < committeesPerSlot; i++ {
-		committee, err := BeaconCommitteeFromState(ctx, state, slot, primitives.CommitteeIndex(i))
-		if err != nil {
-			return nil, err
-		}
+	committeesPerSlot, membersPerCommittee := PtcAllocation(len(committees))
+	for i, committee := range committees {
 		if uint64(len(committee)) < membersPerCommittee {
 			return nil, errCommitteeOverflow
 		}
+		if uint64(i) >= committeesPerSlot {
+			return
+		}
```

Comment:
I think this reads better and you can save one check in the early return case

```suggestion
		if uint64(i) >= committeesPerSlot {
			return
		}
		if uint64(len(committee)) < membersPerCommittee {
			return nil, errCommitteeOverflow
		}
```